libgit/repo.go

```diff
@@ -846,17 +875,37 @@ func GCRepo(
 
 	// Check the GC thresholds again since they might have changed
 	// while getting the lock.
-	doPackRefs, numLooseRefs, err := needsGC(storer, options)
+	doPackRefs, numLooseRefs, doPruneLoose, err := needsGC(storage, options)
 	if err != nil {
 		return err
 	}
-	if !doPackRefs {
+	if !doPackRefs && !doPruneLoose {
 		log.CDebugf(ctx, "GC no longer needed")
 		return nil
 	}
 
-	// For now just pack refs.  TODO: add object pruning and object
-	// re-packing.
-	log.CDebugf(ctx, "Packing %s loose refs", numLooseRefs)
-	return storer.PackRefs()
+	if doPackRefs {
```

Comment:
Wouldn't it be better to first prune loose objects and then pack. I.e. change the order of packing and pruning here.